### PR TITLE
feat: Add Octopus Deploy integration + 2 components

### LINKED ISSUE #2997: [Octopus Deploy] Base
Create a **SuperPlane integration for Octopus Deploy** and **two starter components** (one trigger, one action) so users can build workflows with Octopus Deploy in SuperPlane. This issue covers the base integration and the first components listed below.

**Link**: https://octopus.com/docs

### Connection Method

Part of this task is exploring and determining the best approach to connect Octopus Deploy with SuperPlane.

**Suggested starting point**: Explore Octopus Deploy's docs and the codebase—look at existing integrations and components (e.g. Render, Railway) for patterns.

Consider the available options and evaluate what provides the best balance of security, ease of setup, and functionality for our use cases.

### First Components

‼️ **IMPORTANT**

Creating at least 1 `trigger` and 1 `action` component to test out the Base is required for completing this issue.

Configuration options, data displayed in the payload, and component node meta (details) are for the implementer to suggest based on what the tool offers; you can explore similar components in the codebase for inspiration.

#### 1. On Deployment Event (Trigger)

Runs when a deployment's status changes (e.g. started, failed, completed). Let users filter by event type and by project (and optionally environment) so workflows only run for the events they care about. Enables workflows like notifying Slack or creating a ticket when a deployment fails or succeeds.

#### 2. Deploy Release (Action)

Deploys a chosen release to a chosen environment. Enables workflows like deploying on merge, on approval, or on schedule.

### Acceptance Criteria (for SuperPlane team)

- [ ] integration works
- [ ] starter components work
- [ ] has proper tests
- [ ] has proper documentation
- [ ] passed code quality review
- [ ] passed functionality review
- [ ] passed ui/ux review

### PR DESCRIPTION
**Pull request by zakaneki**
## Summary

Closes #2997

This PR adds a base integration with Octopus Deploy, allowing users to trigger workflows from deployment events and start release deployments directly from SuperPlane.

## Demo

Loom walkthrough: https://www.loom.com/share/acf60dceb634468691c00f8f6a648bb4

### Authentication

* API key authentication via the `X-Octopus-ApiKey` header.
* Supports space-level configuration (custom Space ID or defaults to `Spaces-1`).

### Trigger: On Deployment Event

* Automatically creates Subscriptions in Octopus using the API.
* Uses a unique naming convention (`SuperPlane-{webhookID}`) to avoid collisions between workflow nodes.
* Verifies `X-SuperPlane-Webhook-Secret` to ensure payload authenticity.

### Action: Deploy Release

* Triggers a deployment for a selected Release and Environment.
* Implements a 30s polling mechanism to track deployment status if webhooks fail or are delayed.
* Provides `Success` and `Failed` outputs for conditional workflow branching.

## Technical Decisions

1. **Polling fallback**
   Although Octopus supports task completion webhooks, a polling loop is implemented in the `Deploy Release` action to prevent workflows from hanging if a webhook is dropped. This aligns with other long-running integrations (e.g., Render).

2. **Isolated Subscriptions per trigger node**
   Each "On Deployment Event" node creates its own Subscription. This enables granular filtering (e.g., different environments per node) without cross-event noise.

### Prerequisites

* An Octopus Deploy instance (Cloud or Server).
* `ngrok` for local webhook testing.

### Configure Connection

1. Add a new Octopus connection.
2. Provide:

   * **Server URL** (e.g., `https://example.octopus.app`)
   * **API Key** (generated in Profile → My API Keys)

### Test Trigger

1. Add the "On Deployment Event" node.
2. Create a deployment in Octopus.
3. Verify the workflow triggers.

### Test Action

1. Add the "Deploy Release" node.
2. Select a Project, Release, and Environment.
3. Run the workflow and confirm the deployment starts in Octopus.

### CODE REVIEW

**Comment by AleksandarCole** [comment]
@zakaneki looking good - moved to code review

File: pkg/integrations/octopus/deploy_release.go [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+	var executionCtx *core.ExecutionContext
+	for _, id := range deploymentIDs {
+		found, err := findExecutionByDeploymentID(ctx, id)
+		if err != nil {
+			return okResponse()
+		}
+		if found != nil {
+			executionCtx = found
+			break
+		}
```
Comment (forestileao) [inline pkg/integrations/octopus/deploy_release.go:358]: Is it possible that this may drop a valid match when Octopus sends multiple deployment IDs in one event?

What your tests shows?

File: pkg/integrations/octopus/common.go [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+	if err != nil {
+		return metadata
+	}
+
+	spaceID, err := spaceIDForIntegration(client, integration)
+	if err != nil {
+		return metadata
+	}
+
+	if projectID != "" && !strings.Contains(projectID, "{{") {
```
Comment (forestileao) [inline pkg/integrations/octopus/common.go]: we can make this nested if into a function, right? its the same verification for all 3 resources.
projectId, releaseId, and environmentId

File: pkg/integrations/octopus/deploy_release.go [hunk 3]
```diff
@@ -0,0 +1,470 @@
...
+	"github.com/mitchellh/mapstructure"
+	"github.com/superplanehq/superplane/pkg/configuration"
+	"github.com/superplanehq/superplane/pkg/core"
+)
+
+const (
+	DeployReleasePayloadType          = "octopus.deployment.finished"
+	DeployReleaseSuccessOutputChannel = "success"
+	DeployReleaseFailedOutputChannel  = "failed"
+	DeployReleasePollInterval         = 30 * time.Second
```
Comment (forestileao) [inline pkg/integrations/octopus/deploy_release.go]: Let's increase this to 5 minutes. Like github workflow run and semaphore run workflow. Since we are using polling just as fallback

File: pkg/integrations/octopus/deploy_release.go [hunk 4]
```diff
@@ -0,0 +1,470 @@
...
+	if task.Duration != "" {
+		payload["duration"] = task.Duration
+	}
+
+	return payload
+}
+
+func (c *DeployRelease) Cancel(ctx core.ExecutionContext) error {
+	return nil
+}
```
Comment (forestileao) [inline pkg/integrations/octopus/deploy_release.go:490]: Cancelling the execution should also cancel the deploy on the octopus side

File: pkg/integrations/octopus/deploy_release.go (hunk 1 above)
Comment (zakaneki) [reply pkg/integrations/octopus/deploy_release.go:358]: Yes, it's hard to test in practice because it's an edge case. Definitely an oversight though, thanks!

File: pkg/integrations/octopus/on_deployment_event.go [hunk 5]
```diff
@@ -0,0 +1,428 @@
...
+	}
+	if envIDs, ok := relatedDocs["Environments"]; ok && len(envIDs) > 0 {
+		data["environmentId"] = envIDs[0]
+	}
+	if releaseIDs, ok := relatedDocs["Releases"]; ok && len(releaseIDs) > 0 {
+		data["releaseId"] = releaseIDs[0]
+	}
+	if deploymentIDs, ok := relatedDocs["Deployments"]; ok && len(deploymentIDs) > 0 {
+		data["deploymentId"] = deploymentIDs[0]
+	}
```
[the remark this answers is not in the document]
Comment (zakaneki) [reply pkg/integrations/octopus/on_deployment_event.go:421]: false positive

**Comment by forestileao** [comment]
@zakaneki All good. Tested and everything is working:
image


Thanks for the contribution <3
